src/metadata_serializer.c

```diff
@@ -0,0 +1,119 @@
+#include <string.h>
+#include <errno.h>
+#include <unistd.h>
+#include <endian.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+
+#include "metadata_serializer.h"
+
+struct sedcli_metadata *sedcli_metadata_alloc_buffer()
+{
+	uint8_t *buffer;
+
+	buffer = malloc(SEDCLI_METADATA_SIZE);
+	if (buffer == NULL) {
+		return NULL;
+	}
+
+	memset(buffer, 0, SEDCLI_METADATA_SIZE);
+
+	return (struct sedcli_metadata *)buffer;
+}
+
+void sedcli_metadata_free_buffer(struct sedcli_metadata *buffer)
+{
+	if (buffer != NULL) {
+		memset(buffer, 0, SEDCLI_METADATA_SIZE);
+		free(buffer);
+	}
+}
+
+void sedcli_metadata_init(struct sedcli_metadata *meta, uint32_t pek_id_size,
```

Comment:
starting from this function, I would recommend checking the 'meta' pointer for NULL and in all the future functions as well